pilot/pkg/proxy/envoy/v2/debug.go

```diff
@@ -230,17 +230,22 @@ func (s *DiscoveryServer) distributedVersions(w http.ResponseWriter, req *http.R
 		var results []SyncedVersions
 		adsClientsMutex.RLock()
 		for _, con := range adsClients {
-			con.mu.RLock()
-			if con.node != nil && (proxyNamespace == "" || proxyNamespace == con.node.ConfigNamespace) {
-				// TODO: handle skipped nodes
-				results = append(results, SyncedVersions{
-					ProxyID:         con.node.ID,
-					ClusterVersion:  s.getResourceVersion(con.ClusterNonceAcked, resourceID, knownVersions),
-					ListenerVersion: s.getResourceVersion(con.ListenerNonceAcked, resourceID, knownVersions),
-					RouteVersion:    s.getResourceVersion(con.RouteNonceAcked, resourceID, knownVersions),
-				})
-			}
-			con.mu.RUnlock()
+			// wrap this in independent scope so that panic's don't bypass Unlock...
```

Comment:
won't a panic crash the whole program anyways? or it's just in the handler which catches the panic?

Reply:
yes, just the handler thread panics, but it keeps the read lock open, which is bad.  Theoretically, the string guarding added in this PR should make panic impossible, but I'd rather be safe here, since a panic in any of these routines will stop traffic.

Reply:
you could say the same about 95% of the pilot code though , including every other debug function, which makes me feel like this is not needed here

Reply:
I removed the func wrapper.